pkg/storage/sos/object.go

```diff
@@ -61,9 +63,21 @@ func (c *Client) DeleteObjects(ctx context.Context, bucket, prefix string, recur
 		}
 
 		deleted = append(deleted, res.Deleted...)
+		for _, err := range res.Errors {
+			var e error
+			switch {
+			case err.Message != nil:
+				e = errors.New(*err.Message)
+			case err.Code != nil:
+				e = errors.New(*err.Code)
```

Comment:
Not sure it's a problem, but if it's possible for an error to have a Message AND a Code, this switch will only add the Message AFAIU and the Code would be ignored.

Reply:
Errors are expected to have both Message & Code. Code is intended for machine parsing where Message is human readable error description. We only print errors (we won't parse Code), this is simple fallback in case of exception in SOS so we have some error printed.

Reply:
If we later decide to implement control flow.around Code, we can introduce custom Error. For now this should be enough.